### PR TITLE
Add key as an option to seed

### LINKED ISSUE #103: Pass key instead of seed to algorithms with randomization
If we want this functionality I can implement this.

### PR DESCRIPTION
**Pull request by msluszniak**
Closes #103

### CODE REVIEW

File: lib/scholar/cluster/k_means.ex [hunk 1]
```diff
@@ -73,7 +73,7 @@ defmodule Scholar.Cluster.KMeans do
       """
     ],
     seed: [
-      type: :integer,
+      type: {:custom, Scholar.Options, :seed, []},
```
Comment (polvalente) [inline lib/scholar/cluster/k_means.ex:76]: I think we should fully replace `:seed` with just accepting valid `Nx.Random.key`s

File: lib/scholar/cluster/affinity_propagation.ex [hunk 2]
```diff
@@ -97,11 +97,13 @@ defmodule Scholar.Cluster.AffinityPropagation do
   deftransform fit(data, opts \\ []) do
     opts = NimbleOptions.validate!(opts, @opts_schema)
     opts = Keyword.update(opts, :self_preference, false, fn x -> x end)
-    seed = Keyword.get_lazy(opts, :seed, &System.system_time/0)
-    fit_n(data, seed, NimbleOptions.validate!(opts, @opts_schema))
+    key = Keyword.get_lazy(opts, :key, &System.system_time/0)
+    key_size = Nx.size(key)
+    key = if key_size == 2, do: key, else: Nx.Random.key(Nx.as_type(key, :s64))
```
Comment (josevalim) [inline lib/scholar/cluster/affinity_propagation.ex:102]: Maybe?

```suggestion
    key = Keyword.get_lazy(opts, :key, fn -> Nx.Random.key(System.system_time()) end)
```

File: lib/scholar/cluster/affinity_propagation.ex [hunk 3]
```diff
@@ -40,11 +40,11 @@ defmodule Scholar.Cluster.AffinityPropagation do
...
       doc: "Self preference."
     ],
-    seed: [
-      type: {:custom, Scholar.Options, :seed, []},
+    key: [
+      type: {:custom, Scholar.Options, :key, []},
       doc: """
       Determines random number generation for centroid initialization.
-      If the seed is not provided, it is set to `System.system_time()`.
+      If the key is not provided, it is set to `System.system_time()`.
```
Comment (josevalim) [inline lib/scholar/cluster/affinity_propagation.ex:47]:
```suggestion
      If the key is not provided, it is set to `Nx.Random.key(System.system_time())`.
```

File: lib/scholar/cluster/affinity_propagation.ex (hunk 2 above)
Comment (josevalim) [reply lib/scholar/cluster/affinity_propagation.ex]: We need to apply this diff on the other examples too :)

**Review by josevalim — APPROVED** [review]

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**